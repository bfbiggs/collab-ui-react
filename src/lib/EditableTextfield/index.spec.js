import React from 'react';
import { shallow } from 'enzyme';
import EditableTextField from '@collab-ui/react/EditableTextField';

describe('tests for <EditableTextField />', () => {
  it('should render an Editable Textfield', () => {
    const wrapper = shallow(<EditableTextField />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render one Editable Textfield', () => {
    const container = shallow(<EditableTextField />);

    expect(container.find('span').length).toEqual(1);
    expect(container.find('input').length).toEqual(0);
  });

  it('should click on Editable Textfield', () => {
    const container = shallow(<EditableTextField inputText='Hello World' />);
    container.find('span').simulate('click');

    expect(container.find('span').length).toEqual(0);
    expect(container.find('Input').length).toEqual(1);
  });

  it('should click on Editable Textfield, edit and save', () => {
    const container = shallow(<EditableTextField inputText={'Hello World'}/>);
    container.find('span').simulate('click');
    container.find('Input').simulate('keyDown', { keyCode: 27, key: 'Enter' });

    expect(container.find('span').text()).toEqual('Hello World');
    expect(container.find('span').length).toEqual(1);
    expect(container.find('Input').length).toEqual(0);
  });
});
