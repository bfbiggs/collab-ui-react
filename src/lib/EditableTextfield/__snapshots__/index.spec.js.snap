// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <EditableTextfield /> should render an Editable Textfield 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditableTextfield
    alignment="left"
    buttonClassName=""
    buttonProps={null}
    className=""
    disabled={false}
    handleDoneEditing={null}
    inputText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="cui-editable-textfield__button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          
        </div>,
      ],
      "className": "cui-editable-textfield cui-editable-textfield--left",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "cui-editable-textfield__button",
          "onClick": [Function],
          "onKeyPress": [Function],
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="cui-editable-textfield__button"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            
          </div>,
        ],
        "className": "cui-editable-textfield cui-editable-textfield--left",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "cui-editable-textfield__button",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
